src/pretix/control/templates/pretixcontrol/event/mail.html

```diff
@@ -23,7 +23,29 @@ <h4 class="panel-title">
                     </div>
                     <div id="order_placed" class="panel-collapse collapse">
                         <div class="panel-body">
-                            {% bootstrap_field form.mail_text_order_placed layout="horizontal" %}
+                            <div id="mail_text_order_placed_panel" class="preview-panel" for="mail_text_order_placed">
+                                <label class="col-md-3 control-label">{% trans "Text" %}</label>
+                                <div class="col-md-9">
+                                    <div class="tab-content">
+                                        <div id="mail_text_order_placed_edit" class="tab-pane fade in active">
+                                            {% bootstrap_field form.mail_text_order_placed show_label=False %}
+                                        </div>
+                                        <div id="mail_text_order_placed_preview" class="tab-pane">
+                                            {% for l in request.event.settings.locales %}
+                                                <pre lang="{{ l }}" for="mail_text_order_placed_{{ forloop.counter0 }}"></pre>
+                                            {% endfor %}
+                                        </div>
+                                    </div>
+                                    <ul class="nav nav-pills pull-right">
+                                        <li role="presentation" class="active">
+                                            <a data-toggle="pill" type="edit" href="#mail_text_order_placed_edit"><i class="fa fa-pencil-square-o fa-fw"></i> {% trans "Edit" %}</a>
+                                        </li>
+                                        <li role="presentation">
+                                            <a data-toggle="pill" type="preview" href="#mail_text_order_placed_preview"><i class="fa fa-tv fa-fw"></i> {% trans "Preview" %}</a>
+                                        </li>
+                                    </ul>
+                                </div>
+                            </div>
```

Comment:
Nice idea! Had a quick look at django template engine doc.
`{% include "name_snippet.html" with person="Jane" greeting="Hello" %}`
I think we can probably use this technique to render all the previews with parameter `preview_item`, e.g. `mail_text_order_placed, mail_text_order_paid`.

Or even more aggressive approach, we can use `{% include %}` to avoid writing duplicated `panel` block, i.e.
```
<div class="panel panel-default">
    <div class="panel-heading">
        <h4 class="panel-title">
            <a data-toggle="collapse" href="#order_paid">
                <strong>{% trans "Paid order" %}</strong>
            </a>
        </h4>
    </div>
    <div id="order_paid" class="panel-collapse collapse">
       <div class="panel-body">
           {% bootstrap_field form.mail_text_order_paid layout="horizontal" %}
       </div>
    </div>
</div>
```
The only differences between two mail panels are:
- Panel title, i.e. `{% trans "Paid order" %}`
- Panel id
- form element, i.e. `form.mail_text_order_paid`

What's your opinion? (I'll give a try tomorrow 👍 )